File: app/src/main/java/com/example/architechturestartercode/model/Movie.java
package com.example.architechturestartercode.model;

import com.google.gson.annotations.SerializedName;

public class Movie {
    @SerializedName("id")
    private Long id;
    @SerializedName("title")
    private String title;
    @SerializedName("poster_path")
    private String posterUrl;
    @SerializedName("category")
    private String language;

    public Movie(Long id, String title, String posterUrl, String language) {
        this.id = id;
        this.title = title;
        this.posterUrl = posterUrl;
        this.language = language;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getPosterUrl() {
        return posterUrl;
    }

    public void setPosterUrl(String posterUrl) {
        this.posterUrl = posterUrl;
    }

    public String getLanguage() {
        return language;
    }

    public void setLanguage(String language) {
        this.language = language;
    }
}

File: app/src/main/java/com/example/architechturestartercode/data/movie/model/Movie.kt
package com.example.architechturestartercode.data.movie.model

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import com.google.gson.annotations.SerializedName
const val IMAGE_BASE_URL = "https://image.tmdb.org/t/p/w500"

@Entity(tableName = "movies")
data class Movie(
    @SerializedName("id")
    @ColumnInfo(name = "id")
    @PrimaryKey
    var id: Long,

    @SerializedName("original_title")
    @ColumnInfo(name = "title")
    var title: String,

    @SerializedName("poster_path")
    @ColumnInfo(name = "poster_url")
    var posterUrl: String,

    @SerializedName("original_language")
    @ColumnInfo(name = "language")
    var language: String
) {
    val fullPosterUrl: String
        get() = IMAGE_BASE_URL + posterUrl
}

